### ISSUE TITLE
Design Intent - Accessibility Feedback - MHV medications list filter

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#90241]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Lots of stuff that I like here!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [ ] **Must:** For the group of checkboxes, there needs to be a clearly-stated question or prompt to provide sufficient context for the checkboxes. In the `va-checkbox-group` component, this is the `label` prop, which maps to a `legend` for a `fieldset` surrounding all of the checkboxes (see [Storybook examples](https://design.va.gov/storybook/?path=/docs/uswds-va-checkbox-group--docs). Something like "Select types of medications to display" or similar would be fine for this prompt. (If you end up changing the filter behavior or language around filtering based on Matt's feedback, you'll probably want to do something different!)
   
   This is a little duplicative of the text in the accordion toggle, but it benefits screen reader users significantly. This question or prompt is programmatically associated with the checkboxes, meaning it will always be announced when a screen reader user tabs to the first checkbox in the group. Where a sighted user might see that there's only one grouping of checkboxes and infer that they should choose the ones they want to display, a screen reader user may not get those visual cues. Is this checkbox the first in a single group, or are there multiple groups of checkboxes which each control different filtering behavior? It wouldn't take too long exploring the page to put together a clear mental model, but we don't want to force users to do extra exploring when we can just provide immediate context with a clear prompt.
- [ ] **Should:** There was some discussion from pretty much all of the practice areas about the ordering of elements in the "Showing X to Y" - "Show other medications" - "Showing these medications" sequence. I'm going to make a recommendation that I think is consistent with what others said and is consistent with Erin's feedback in the IA ticket.
   
   Here's how I'd like to see it ordered:
   - H2: Medications list
   - Accordion (with [accordion item `level`](https://design.va.gov/storybook/?path=/docs/uswds-va-accordion--docs#custom-header-level) set to 3 to make it an H3): **Select types of medications to display** (or something like that, since "other" doesn't make sense with where it's placed here)
   - H3 (styled as regular text): Showing 1-20 of 51 **active** medications, sorted A to Z by name (consistent with Erin's recommendation)
   - Medications list
   
   Here's what that gets us:
   - For screen reader users navigating by heading, there's an easy way to find (1) the section of the page with the medications list, (2) the filter options, and (2) the list itself with context about what's being shown.
   - When applying the filters, keyboard focus should move to the "Showing 1-20" heading. Moving keyboard focus ensures that screen reader users hear this text immediately --- which gives them confirmation that the filters have been applied and a concise status update about what's now displayed on the page.
   
   This isn't the only way to do it, but I think it would be the most screen reader-friendly approach and consistent with what some other teams are doing elsewhere on VA.gov. But if you have concerns please feel free to ping me in Slack!

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [ ] Add the VFS team product label
- [ ] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [ ] Add the Collaboration Cycle initiative milestone

**Comment by alexiawunder** [comment]
1. **For the group of checkboxes, there needs to be a clearly-stated question or prompt:** We’ve worked with the content team to come up with a prompt for the checkboxes. 
2. **There was some discussion from pretty much all of the practice areas about the ordering of elements in the "Showing X to Y" - "Show other medications":** We initially separated it to inform users about what was being displayed before they opened the accordion, and to prevent the focus state from landing above the accordion when it closed. Now that we're shifting away from the 'reverse filter' model, we will place a single medication count below the accordion.
